python3Packages.hdfs: init at 2.5.8

samuela opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change
Add the hdfs python package.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

samuela commented:
Update: CI is currently failing because I forgot this depends on https://github.com/NixOS/nixpkgs/pull/137685. Once that lands, I'll rebase this and we should be good to go.

fabaff commented on pkgs/development/python-modules/hdfs/default.nix:
```diff
@@ -0,0 +1,34 @@
+  checkInputs = [ fastavro nose pytestCheckHook ];
+
+  meta = with lib; {
+    description = "HdfsCLI: API and command line interface for HDFS.";
```
```suggestion
    description = "Python API and command line interface for HDFS";
```

No period at the end -> see https://nixos.org/manual/nixpkgs/stable/#var-meta-description

If a heavy use case is the command line then it should also go to `all-packages.nix` with `toPythonApplication`.

samuela replied:
Ah, thanks for that link! I wasn't aware of those style guidelines

fabaff reviewed (commented):
Could you add `pythonImportsCheck` as additional safe-guard?

samuela commented:
> Could you add `pythonImportsCheck` as additional safe-guard?

Done in https://github.com/NixOS/nixpkgs/pull/137713/commits/efe93cf0fd2fefc4d7ac1c9c71ccaa63591ba50e

samuela committed:
python3Packages.hdfs: init at 2.5.8

samuela commented:
I rebased and squashed on the latest master now that https://github.com/NixOS/nixpkgs/pull/137685 has been merged. I think this should be ready to go now!

smancill commented:
Fails to build on Darwin because of #137678:

```
 nixpkgs-review pr 137713
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/137713/head:refs/nixpkgs-review/1
remote: Enumerating objects: 9, done.
remote: Counting objects: 100% (6/6), done.
remote: Total 9 (delta 5), reused 5 (delta 5), pack-reused 3
Unpacking objects: 100% (9/9), 46.71 KiB | 810.00 KiB/s, done.
From https://github.com/NixOS/nixpkgs
 + 2dfffe79b2e...a4d96cd808f master                -> refs/nixpkgs-review/0  (forced update)
 + b93b2a2c21f...e3ae16550b1 refs/pull/137713/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /Users/smancill/.cache/nixpkgs-review/pr-137713/nixpkgs a4d96cd808f13d924988938ecf9a46b2a2787fa3
Preparing worktree (detached HEAD a4d96cd808f)
Updating files: 100% (28104/28104), done.
HEAD is now at a4d96cd808f Merge pull request #139674 from mweinelt/python/clevercsv
$ nix-env --option system x86_64-darwin -f /Users/smancill/.cache/nixpkgs-review/pr-137713/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit --no-ff e3ae16550b1a49fba50fea1a225637217f8b5a35
Auto-merging pkgs/top-level/python-packages.nix
Automatic merge went well; stopped before committing as requested
$ nix-env --option system x86_64-darwin -f /Users/smancill/.cache/nixpkgs-review/pr-137713/nixpkgs -qaP --xml --out-path --show-trace --meta
2 packages added:
python38Packages.hdfs (init at 2.5.8) python39Packages.hdfs (init at 2.5.8)

$ nix --experimental-features nix-command build --no-link --keep-going -f /Users/smancill/.cache/nixpkgs-review/pr-137713/build.nix
error: builder for '/nix/store/ib9vjgbxjah5668y0wbx9d7v3mspjzx5-python3.8-beautifulsoup4-4.10.0.drv' failed with exit code 1;
       last 10 log lines:
       >         soup = self.soup(markup)
       > >       self.assertEqual('Sacr\xe9 bleu!', soup.root.string)
       > E       AttributeError: 'NoneType' object has no attribute 'string'
       >
       > bs4/testing.py:1002: AttributeError
       > =========================== short test summary info ============================
       > FAILED bs4/tests/test_lxml.py::LXMLTreeBuilderSmokeTest::test_real_hebrew_document
       > FAILED bs4/tests/test_lxml.py::LXMLTreeBuilderSmokeTest::test_smart_quotes_converted_on_the_way_in
       > FAILED bs4/tests/test_lxml.py::LXMLXMLTreeBuilderSmokeTest::test_can_parse_unicode_document
       > ======================== 3 failed, 514 passed in 1.97s =========================
       For full logs, run 'nix log /nix/store/ib9vjgbxjah5668y0wbx9d7v3mspjzx5-python3.8-beautifulsoup4-4.10.0.drv'.
error: 1 dependencies of derivation '/nix/store/gfy98yssd5q9k31acbd6szmn6j3457qp-python3.8-pandas-1.3.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/qa29p8xzl0rg7fjb2k8jbb4blkq29bgx-python3.8-fastavro-1.4.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vi21p4zb57xgjc16ifq32z6xliih6ryk-python3.8-hdfs-2.5.8.drv' failed to build
error: builder for '/nix/store/vq9nm18v7qp5mjpg1487qnjpwq4pmqsm-python3.9-beautifulsoup4-4.10.0.drv' failed with exit code 1;
       last 10 log lines:
       >         soup = self.soup(markup)
       > >       self.assertEqual('Sacr\xe9 bleu!', soup.root.string)
       > E       AttributeError: 'NoneType' object has no attribute 'string'
       >
       > bs4/testing.py:1002: AttributeError
       > =========================== short test summary info ============================
       > FAILED bs4/tests/test_lxml.py::LXMLTreeBuilderSmokeTest::test_real_hebrew_document
       > FAILED bs4/tests/test_lxml.py::LXMLTreeBuilderSmokeTest::test_smart_quotes_converted_on_the_way_in
       > FAILED bs4/tests/test_lxml.py::LXMLXMLTreeBuilderSmokeTest::test_can_parse_unicode_document
       > ======================== 3 failed, 514 passed in 2.83s =========================
       For full logs, run 'nix log /nix/store/vq9nm18v7qp5mjpg1487qnjpwq4pmqsm-python3.9-beautifulsoup4-4.10.0.drv'.
error: 1 dependencies of derivation '/nix/store/j7g7l6q0phgj59jza1csaqv1as1pzl53-python3.9-pandas-1.3.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/rwry5hxb034skjn7xqzqp3a7w6n6d29d-python3.9-fastavro-1.4.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/dwzg5i8f7jy14cq6y51ixj61p8lyzxym-python3.9-hdfs-2.5.8.drv' failed to build
error: 2 dependencies of derivation '/nix/store/2h41fl995n06mglijfp4hndb3lcg22nj-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/137713

2 packages failed to build:
python38Packages.hdfs python39Packages.hdfs

error: build log of '/nix/store/vi21p4zb57xgjc16ifq32z6xliih6ryk-python3.8-hdfs-2.5.8.drv' is not available
error: build log of '/nix/store/8ymz34hb8ckfc3v05zlwq78akvamrc3y-python3.8-hdfs-2.5.8' is not available
error: build log of '/nix/store/dwzg5i8f7jy14cq6y51ixj61p8lyzxym-python3.9-hdfs-2.5.8.drv' is not available
error: build log of '/nix/store/05957yw0fi56fpwgi842bj5sfgzlbsz4-python3.9-hdfs-2.5.8' is not available
```

SuperSandro2000 commented:
@ofborg eval
@ofborg build python39Packages.hdfs
